app/views/gitlab_oauth.py

```diff
@@ -0,0 +1,58 @@
+import datetime
+import os
+
+from dotenv import load_dotenv
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework_simplejwt.tokens import RefreshToken
+
+from ..oauth import generate_gitlab_access_token, retrieve_gitlab_user_info
+
+from ..models import User
+from ..serializers import UserSerializer
+
+load_dotenv()
+
+# GITHUB ID AND SECRET
+SOCIAL_AUTH_GITLAB_KEY = os.getenv("SOCIAL_AUTH_GITLAB_KEY")
+SOCIAL_AUTH_GITLAB_SECRET = os.getenv("SOCIAL_AUTH_GITLAB_SECRET")
```

Comment:
Declare these values where they are needed, check that they exist before using them.